R/headers.R

```diff
@@ -0,0 +1,21 @@
+#' Add headers to endpoint output data
+#'
+#' Intended to be used from endpoint target function. Note `Content-Type`
+#' headers are handled by returning arg to endpoint.
+#'
+#' @param data Response data
+#' @param headers Named list of headers to add.
+#'
+#' @return Data from endpoint target with headers
+#' @export
+#'
+#' @examples
+#' pkgapi_add_headers("output", list("Content-Dispotition" = "output_file.txt"))
+pkgapi_add_headers <- function(data, headers) {
+  attributes(data) <- list("headers" = headers)
+  data
+}
+
+headers <- function(data) {
+  attr(data, "headers")
```

Comment:
I think that using "pkgapi_headers" as the attribute name will protect us from any possible collision - you might change the name of this function to be `get_pkgapi_headers` too? (That will avoid clashes with `httr::headers` for example).

Also please use `attr(..., exact = TRUE)` for attr to avoid partial matching